sql/sql_yacc.yy

```diff
@@ -13405,7 +13405,8 @@ delete_single_table:
               MYSQL_YYABORT;
             YYPS->m_lock_type= TL_READ_DEFAULT;
             YYPS->m_mdl_type= MDL_SHARED_READ;
-            Lex->last_table()->period_conditions= Lex->period_conditions;
+            if ($3)
+              Lex->last_table()->period_conditions= Lex->period_conditions;
```

Comment:
I can't figure out, where or when is the place for reinitialization (or is it here, maybe)

Besides, this is the same behavior as in `table_primary_ident` definition -- just not to copy `Lex->period_conditions` if no conditions were specified. BTW, it means, that Lex will store the last specified period, and it can be used (though its unclear) to determine if at least one period condition was specified in the query

Reply:
True, it is the way `Lex` used. There is no reinit, just forgot about it.